pkg/operation/botanist/coredns.go

```diff
@@ -41,17 +41,27 @@ func (b *Botanist) DefaultCoreDNS() (coredns.Interface, error) {
 	values := coredns.Values{
 		// resolve conformance test issue (https://github.com/kubernetes/kubernetes/blob/master/test/e2e/network/dns.go#L44)
 		// before changing
-		ClusterDomain:   gardencorev1beta1.DefaultDomain,
-		ClusterIP:       b.Shoot.Networks.CoreDNS.String(),
-		Image:           image.String(),
-		PodNetworkCIDR:  b.Shoot.Networks.Pods.String(),
-		NodeNetworkCIDR: b.Shoot.GetInfo().Spec.Networking.Nodes,
+		ClusterDomain:                     gardencorev1beta1.DefaultDomain,
+		ClusterIP:                         b.Shoot.Networks.CoreDNS.String(),
+		Image:                             image.String(),
+		PodNetworkCIDR:                    b.Shoot.Networks.Pods.String(),
+		NodeNetworkCIDR:                   b.Shoot.GetInfo().Spec.Networking.Nodes,
+		UseClusterProportionalAutoscaling: false,
 	}
 
 	if b.APIServerSNIEnabled() {
 		values.APIServerHost = pointer.String(b.outOfClusterAPIServerFQDN())
 	}
 
+	if b.Shoot.GetInfo().Spec.SystemComponents != nil && b.Shoot.GetInfo().Spec.SystemComponents.CoreDNS != nil && b.Shoot.GetInfo().Spec.SystemComponents.CoreDNS.Autoscaling.Mode == gardencorev1beta1.CoreDNSAutoscalingModeClusterProportional {
```

Comment:
Nit: You could introduce a helper function in `pkg/apis/core/v1beta1/helper` to retrieve the CoreDNS autoscaling mode when providing the `*SystemComponents`.

Reply:
Extracted a helper function as suggested.